deepchecks/vision/dataset.py

```diff
@@ -45,11 +46,9 @@ class VisionDataset:
     ----------
     data_loader : DataLoader
         PyTorch DataLoader object. If your data loader is using IterableDataset please see note below.
-    label_type : str
-        Type of label. Must be one of the following: 'classification', 'object_detection'.
     num_classes : int, optional
         Number of classes in the dataset. If not provided, will be inferred from the dataset.
-    label_transformer : Callable, optional
+    label_transformer : Union[ClassificationLabelEncoder, DetectionLabelEncoder]
```

Comment:
Now that I see it, won't it be confusing with scikit-learn's LabelEncoder? @noamzbr WDYT?

Reply:
I agree it's confusing. any idea for a better name?

Reply:
label formatter
label translator
label molder 
label schema

only first two are real candidates but you never know...